concurrency-tests/src/jcstress/java/io/micrometer/concurrencytest/HistogramConsistencyTest.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright 2022 VMware, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micrometer.concurrencytest;
+
+import io.micrometer.core.instrument.Clock;
+import io.micrometer.core.instrument.distribution.CountAtBucket;
+import io.micrometer.core.instrument.distribution.DistributionStatisticConfig;
+import io.micrometer.core.instrument.distribution.Histogram;
+import io.micrometer.core.instrument.distribution.TimeWindowFixedBoundaryHistogram;
+import org.openjdk.jcstress.annotations.*;
+import org.openjdk.jcstress.infra.results.DDD_Result;
+
+@JCStressTest
+@Outcome(id = "0.0, 0.0, 0.0", expect = Expect.ACCEPTABLE, desc = "read before all writes")
+@Outcome(id = "1.0, 1.0, 2.0", expect = Expect.ACCEPTABLE, desc = "write all before read")
+@Outcome(id = "1.0, 1.0, 1.0", expect = Expect.ACCEPTABLE, desc = "read after first write")
+@Outcome(id = "0.0, 0.0, 1.0", expect = Expect.FORBIDDEN, desc = "read first write mid snapshot")
```

Comment:
With the changes in #3394, this case can still happen, but the current description isn't quite accurate. I believe with those changes, this case is when both recordings happen between the second and third bucket being read.